src/map/entities/mobentity.h

```diff
@@ -57,7 +57,7 @@ enum ROAMFLAG : uint16
     ROAMFLAG_NONE2   = 0x04,  //
     ROAMFLAG_NONE3   = 0x08,  //
     ROAMFLAG_NONE4   = 0x10,  //
-    ROAMFLAG_NONE5   = 0x20,  //
+    ROAMFLAG_SCRIPT  = 0x20,  // Will be implemented in a lua script (e.g. entity.onMobRoamAction())
```

Comment:
> ```lua
> entity.onMobRoam = function(mob)
> ```

I didn't use this one because it's hard coded to run once every three seconds. The `onPath()` one can never avoid the code that forces a mob to roam home, which would make `isFollowingPath()` true while roaming home.

Reply:
> I didn't use this one because it's hard coded to run once every three seconds. The `onPath()` one can never avoid the code that forces a mob to roam home, which would make `isFollowingPath()` true while roaming home.

If thats all then we can use a different condition to not re-kick off that function till after it is within a range from home again. For example, a `mob:getLocalVar()` can be read and only kick off pathing if zero, and it within 5 yalms of its homepoint, set it back to zero again. in onPath could set it to one because one in there its pathing.

We could also store the last position it had before it was in battle mode, and path functions do include having it go back to its last node - we just make that pos its last node. Those I do not know how well that works - we haven't been using it except in big tabled NMs.

Reply:
As we discussed in DMs, I'll make a MobMod that disables the behavior that forces a mob to roam home when it gets a set distance from its spawn location. Then I could refactor the mobs currently using ROAMFLAG_EVENT (dynamis and battlefield mobs) so they use lua to face their spawn location after roaming home (e.g. `onMobRoam() to check distance from spawn and reface its spawn facing when it's 0). Then claim ROAMFLAG_EVENT for my purposes.

I added the `triggerListener()` stuff because I wanted `xi.follow.follow()` to be a fire and forget function, so it could be used in, say, a GM command for example.

Reply:
In addition to `DONT_ROAM_HOME` I created one more MobMod: `ROAM_RESET_FACING` which automatically gets set for all the dynamis and battlefield mobs that were relying on `ROAMFLAG_EVENT` to do it before.